Makefile

```diff
@@ -1,13 +1,21 @@
-# Run the protoc compiler to generate the Golang server code.
-codegen: update-submodules
-	echo `pwd`
-	docker run --rm -v `pwd`/grpc:/app/grpc -v `pwd`/misc:/app/misc -w /app noted-python-protoc /bin/sh -c misc/gen_proto.sh
+# Init submodules
+init-submodules:
+	git submodule init
 
 # Fetch the latest version of the protos submodule.
 update-submodules:
 	git submodule update --remote
 
-# After cloning the repo, run init
-init:
-	git submodule init
-	docker build -t noted-python-protoc -f misc/Dockerfile .
+# Builds a Docker image that'll run the recommendations service
+service-build: init-submodules update-submodules
+	docker build -t noted-recommendations-service -f Dockerfile .
+
+
+# placeholder
+service-run:
+	docker run noted-recommendations-service -d -p 8080:8080
+# placeholder
+
+
+
+
```

Comment:
Same thing, you should configure your editor to trim excessive newlines at the end of a file.